app/models/ndr_error/log.rb

```diff
@@ -32,9 +32,20 @@ class Log < ActiveRecord::Base
     before_create :register_system
     before_create :set_uuid_primary_key
 
+    def self.text_columns
+      user_column = NdrError.user_column.to_s
+      %w(error_class description).tap do |text_columns|
+        # Allow searching of `user_column` if it is textual:
+        if :string == columns_hash[user_column].try(:type)
+          text_columns << user_column
+        else
+          fail SecurityError, "Column '#{user_column}' not found!"
+        end
+      end
+    end
+
     def self.filter_by_keywords(keywords)
-      columns    = %w(description error_class) << NdrError.user_column.to_s
-      fragment   = columns.map { |column| "lower(#{column}) LIKE lower(:key)" }.join(' OR ')
+      fragment   = text_columns.map { |column| "lower(#{column}) LIKE lower(:key)" }.join(' OR ')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.